libsolidity/ast/Types.h

```diff
@@ -709,13 +716,18 @@ class CompositeType: public Type
 	/// Each type is included only once.
 	std::vector<Type const*> fullDecomposition() const;
 
+	DataLocation location() const { return m_location; }
+
```

Comment:
Why this change? Seems unrelated to the PR. Is it just a refactor because it makes more sense inside `CompositeType` or is there something more to it?

BTW, I think you could extract this and the other tiny refactors (`auto` -> `MemberList::Member`, docstring tweaks) into a separate PR to get them out of the way since they're just noise here and could be probably merged right away.

Reply:
`CompositeType::location()` is used [here](https://github.com/ethereum/solidity/pull/9746/files#diff-9f2a45efaa7b75ab7ec620d3177cf01aR618).

Regarding other minor refactoring, I am not sure if we rich the critical mass here, but in general it is a good idea.

Reply:
Oh, right, somehow I missed that. It's fine then.